File: core-banking/backend/module/domain/src/main/java/com/revy/example/domain/post/Post.java
package com.revy.example.domain.post;

import com.revy.example.domain.common.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.Instant;

@Entity
@Table(name = "post")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter(AccessLevel.PUBLIC)
public class Post extends BaseEntity {

    @Column(name = "user_id")
    private Long userId;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(nullable = false, length = 500)
    private String summary;

    @Column(nullable = false, columnDefinition = "text")
    private String content;

    @Column(nullable = false, length = 50)
    private String category;

    @Column(nullable = false, length = 20)
    private String status;

    @Column(nullable = false)
    private Long viewCount;

    @Column(name = "published_at")
    private Instant publishedAt;

    public static Post create(Long userId, String title, String summary, String content) {
        Post post = new Post();
        post.userId = userId;
        post.title = title;
        post.summary = summary;
        post.content = content;
        post.category = "GENERAL";
        post.status = "PUBLISHED";
        post.viewCount = 0L;
        post.publishedAt = Instant.now();
        return post;
    }
}

File: code-playbook/practice_kotlin/kotlin_practice_2/src/main/kotlin/com/revy/domain/Post.kt
package com.revy.domain

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.util.UUID
import kotlin.uuid.ExperimentalUuidApi
import kotlin.uuid.Uuid
import kotlin.uuid.toJavaUuid

@Entity
@Table(name="post")
class Post {
    @OptIn(ExperimentalUuidApi::class)
    @Id
    @Column(name="id", columnDefinition = "uuid")
    val id: UUID = Uuid.generateV7().toJavaUuid()

    @Column(name="title")
    var title: String = ""

    @Column(name="content")
    var content: String = ""

    companion object {
        fun createNewPost(title: String, content: String): Post {
            var newPost = Post()
            newPost.title = title
            newPost.content = content
            return newPost

        }
    }
}